test/xlsxtream/io/zip_tricks_io_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+require 'xlsxtream/workbook'
+require 'xlsxtream/io/hash'
+require 'zip'
+
+module Xlsxtream
+  class ZipTricksIOTest < Minitest::Test
+
+    def test_writes_of_multiple_files
+      zip_buf = Tempfile.new('ztio-test')
+
+      io = Xlsxtream::IO::ZipTricksIO.new(zip_buf)
+      io.add_file("book1.xml")
+      io << '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><workbook />'
+      io.add_file("book2.xml")
+      io << '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><workbook>'
+      io << '</workbook>'
+      io.add_file("empty.txt")
+      io.add_file("another.xml")
+      io << '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><another />'
+      io.close
+
+      zip_buf.rewind
+
+      files_contents = {}
+      ::Zip::File.open(zip_buf.path) do |zip_file|
```

Comment:
Please pass the `zip_buf` to `Zip::File.open`, otherwise there is no point in rewinding the IO above.